service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/UpdateNotebookCpuAndGpuAttributesStep.java

```diff
@@ -0,0 +1,69 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.ainotebook;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.UPDATE_ACCELERATOR_CONFIG;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.UPDATE_MACHINE_TYPE;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.DbSerDes;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.Optional;
+
+// Update the CPU and GPU attributes in the database.
+public class UpdateNotebookCpuAndGpuAttributesStep implements Step {
```

Comment:
I think the code would be simpler if you combined the UpdateAiNotebookMetadataAttributesStep and this step into a single update of the attribute data. I don't think there is a reason the metadata update has to come before updating the CPU/GPU.